### PR TITLE
feat(ingestion): Adds --dry-run and --preview options to datahub ingest command.

### PR DESCRIPTION
**Pull request by rslanka**
Adds `--dry-run` and `--preview` options to `datahub ingest` command.

The `--dry-run` option of the `ingest` command performs all of the ingestion steps, skipping the writing to the sink step, producing the standard run summary.

The `--preview` option of the `ingest` command performs all of the standard ingestion steps, except that it restricts processing to at most 10 work units.

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: metadata-ingestion/src/datahub/cli/ingest_cli.py [hunk 1]
```diff
@@ -40,7 +40,21 @@ def ingest() -> None:
     help="Config file in .toml or .yaml format.",
     required=True,
 )
-def run(config: str) -> None:
+@click.option(
+    "--dry-run",
```
Comment (shirshanka) [inline metadata-ingestion/src/datahub/cli/ingest_cli.py:44]: add `-n` short form to align with other unix tools?

File: metadata-ingestion/src/datahub/cli/ingest_cli.py [hunk 2]
```diff
@@ -40,7 +40,21 @@ def ingest() -> None:
...
+    type=bool,
+    is_flag=True,
+    default=False,
+    help="Perform a dry run of the ingestion, essentially skipping writing to sink.",
+)
+@click.option(
+    "--preview-mode",
```
Comment (shirshanka) [inline metadata-ingestion/src/datahub/cli/ingest_cli.py:51]: rename to `--preview`  (-mode is redundant I think)

File: metadata-ingestion/src/datahub/cli/ingest_cli.py [hunk 3]
```diff
@@ -40,7 +40,21 @@ def ingest() -> None:
...
+    type=bool,
+    is_flag=True,
+    default=False,
+    help="Perform limited ingestion from the source to the sink to get a quick preview.",
+)
+def run(config: str, dry_run: bool, preview_mode: bool) -> None:
```
Comment (shirshanka) [inline metadata-ingestion/src/datahub/cli/ingest_cli.py:57]: if you change the option name from preview_mode to preview, you might have to change this variable name as well

File: metadata-ingestion/src/datahub/ingestion/run/pipeline.py [hunk 4]
```diff
@@ -131,21 +137,26 @@ def _configure_transforms(self) -> None:
...
         callback = LoggingCallback()
         extractor: Extractor = self.extractor_class()
-        for wu in self.source.get_workunits():
+        for wu in itertools.islice(
+            self.source.get_workunits(), self.max_workunits_to_process
+        ):
             # TODO: change extractor interface
             extractor.configure({}, self.ctx)
 
             self.sink.handle_work_unit_start(wu)
```
Comment (shirshanka) [inline metadata-ingestion/src/datahub/ingestion/run/pipeline.py:155]: should this method also be protected by the `--dry-run` check?

File: metadata-ingestion/src/datahub/ingestion/run/pipeline.py [hunk 5]
```diff
@@ -131,21 +137,26 @@ def _configure_transforms(self) -> None:
...
             record_envelopes = extractor.get_records(wu)
             for record_envelope in self.transform(record_envelopes):
-                self.sink.write_record_async(record_envelope, callback)
+                if not self.dry_run:
+                    self.sink.write_record_async(record_envelope, callback)
 
             extractor.close()
             self.sink.handle_work_unit_end(wu)
```
Comment (shirshanka) [inline metadata-ingestion/src/datahub/ingestion/run/pipeline.py:162]: Should this method invocation also be protected by the `--dry-run` check?

**Review by shirshanka — CHANGES_REQUESTED** [review]
Some changes requested. 
Pretty clean PR otherwise!

File: metadata-ingestion/src/datahub/ingestion/run/pipeline.py (hunk 4 above)
Comment (rslanka) [reply metadata-ingestion/src/datahub/ingestion/run/pipeline.py:155]: Nope! We can do dry-run outside of preview mode as well (where `self.max_workunits_to_process` will be `None` and we get all workunits) .

File: metadata-ingestion/src/datahub/ingestion/run/pipeline.py (hunk 5 above)
Comment (rslanka) [reply metadata-ingestion/src/datahub/ingestion/run/pipeline.py:162]: It already is, right?

**Review by shirshanka — APPROVED** [review]
LGTM

**Pull request merged**